ktor-shared/ktor-serialization/ktor-serialization-jackson/jvm/src/JacksonConverter.kt

```diff
@@ -95,6 +131,61 @@ public class JacksonConverter(
             }
         }
     }
+
+    private val prettyPrinting by lazy {
+        objectMapper.isEnabled(SerializationFeature.INDENT_OUTPUT)
+    }
+
+    private companion object {
+        private const val beginArrayCharCode = '['.code
+        private const val endArrayCharCode = ']'.code
+        private const val objectSeparator = ','.code
+    }
+
+    /**
+     * Guaranteed to be called inside a [Dispatchers.IO] context, see [OutputStreamContent]
+     */
+    @Suppress("BlockingMethodInNonBlockingContext")
+    private suspend fun <T> Flow<T>.serializeJson(outputStream: OutputStream) {
+        val jfactory = JsonFactory()
+        // cannot use ObjectMapper write to Stream because it flushes the OutputStream
```

Comment:
can you elaborate on flushing here and why is it bad?

Reply:
I changed it to a manual flush after each value + separator (or endChar for the last value) in Flow, as you suggested for Gson, and made this behavior consistent in all 3 tools : Gson, Jackson and kotlinx-serialization